larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.cc

```diff
@@ -0,0 +1,99 @@
+/**
+ *  @file   larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.cc
+ *
+ *  @brief  Implementation file for the cheated threeD reclustering algorithm.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+#include "larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.h"
+
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+CheatedThreeDClusteringTool::CheatedThreeDClusteringTool()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool CheatedThreeDClusteringTool::Run(const Algorithm *const pAlgorithm, std::vector<pandora::CaloHitList*> &newCaloHitListsVector)
+{
+    if (newCaloHitListsVector.empty() || newCaloHitListsVector.size() != 1)
+        return false;
+
+    CaloHitList *initialCaloHitList = newCaloHitListsVector.at(0);
+    newCaloHitListsVector.clear();
+
+    std::map<int, CaloHitList*> McIdCaloHitListMap;
+
+    for (const CaloHit *const pCaloHit : *initialCaloHitList)
+    {
+        const CaloHit *const pParentCaloHit = static_cast<const CaloHit *>(pCaloHit->GetParentAddress());
+
+        int mainMcParticleIndex = this->GetMainMcParticleIndex(pAlgorithm, pParentCaloHit);
+
+        if (McIdCaloHitListMap.find(mainMcParticleIndex) != McIdCaloHitListMap.end()) 
+        {
+            McIdCaloHitListMap.at(mainMcParticleIndex)->push_back(pCaloHit);
+        }
+        else
+        {
+            CaloHitList* newList = new CaloHitList();
+            McIdCaloHitListMap.insert(std::make_pair(mainMcParticleIndex, newList));
+        } 
+
+    }
+    for (const auto& pair : McIdCaloHitListMap) newCaloHitListsVector.push_back(pair.second);
```

Comment:
Statement inside the for loop should move to the following line?

Reply:
I take it that you are just suggesting fixing the formatting here - in which case, will move it.